cmd/bucket-handlers.go

```diff
@@ -971,9 +970,30 @@ func (api objectAPIHandlers) PostPolicyBucketHandler(w http.ResponseWriter, r *h
 
 		var b bytes.Buffer
 		if fileName == "" {
+			if http.CanonicalHeaderKey(name) == http.CanonicalHeaderKey("x-minio-fanout-list") {
+				dec := json.NewDecoder(part)
+				dec.DisallowUnknownFields()
```

Comment:
This is a double edged sword. I tend to agree that we want older versions to complain about unknown fields.

Reply:
Yeah this is mostly defensive, can be removed.

Reply:
I think it is reasonable to keep it. It is actually a good thing.

How about if we do something like this:

```
type PutObjectFanOutRequest struct {
	Key                string            `json:"key"`
	...
	Optional       map[string]string `json:"optional,omitempty"`
}
```

This will allow us to add optional values, that can safely be ignored, and we reject all other unknown fields?

Reply:
Its okay let me remove it.